src/accounts/avalanche.ts

```diff
@@ -0,0 +1,118 @@
+import shajs from "sha.js";
+import { Account } from "./account";
+import { GetVerificationBuffer } from "../messages";
+import { BaseMessage, Chain } from "../messages/message";
+import { decrypt as secp256k1_decrypt, encrypt as secp256k1_encrypt } from "eciesjs";
+import { KeyPair } from "avalanche/dist/apis/avm";
+import { Avalanche, BinTools, Buffer as AvaBuff } from "avalanche";
+
+/**
+ * AvalancheAccount implements the Account class for the Avalanche protocol.
+ * It is used to represent an Avalanche account when publishing a message on the Aleph network.
+ */
+export class AvalancheAccount extends Account {
+    private signer;
+
+    constructor(signer: KeyPair) {
+        super(signer.getAddress().toString("hex"), signer.getPublicKey().toString("hex"));
+        this.signer = signer;
+    }
+
+    override GetChain(): Chain {
+        return Chain.AVAX;
+    }
+
+    /**
+     * Encrypt a content using the user's public key from the keypair
+     *
+     * @param content The content to encrypt.
+     */
+    encrypt(content: Buffer): Buffer {
+        return secp256k1_encrypt(this.publicKey, content);
+    }
+
+    /**
+     * Decrypt a given content using the private key from the keypair.
+     *
+     * @param encryptedContent The encrypted content to decrypt.
+     */
+    decrypt(encryptedContent: Buffer): Buffer {
+        const secret = this.signer.getPrivateKey().toString("hex");
+        return secp256k1_decrypt(secret, encryptedContent);
+    }
+
+    private async digestMessage(message: Buffer) {
+        const msgSize = Buffer.alloc(4);
+        msgSize.writeUInt32BE(message.length, 0);
+        const msgStr = message.toString("utf-8");
+        const msgBuf = Buffer.from(`\x1AAvalanche Signed Message:\n${msgSize}${msgStr}`, "utf8");
+
+        return new shajs.sha256().update(msgBuf).digest();
+    }
+
+    /**
+     * The Sign method provides a way to sign a given Aleph message using an avalanche keypair.
+     * The full message is not used as the payload, only fields of the BaseMessage type are.
+     *
+     * The sign method of the keypair is used as the signature method.
+     *
+     * @param message The Aleph message to sign, using some of its fields.
+     */
+    async Sign(message: BaseMessage): Promise<string> {
+        const buffer = GetVerificationBuffer(message);
+        const digest = await this.digestMessage(buffer);
```

Comment:
why are you additionally running digestMessage here? Every other chains only uses GetVerificationBuffer and runs a signature over it.

Reply:
It mirrors the [functionalities of the JS library](https://github.com/aleph-im/aleph-js/blob/master/src/api/avalanche.js#L49) where the message if prefixed with the chain that issued it. The main idea behind this is to prevent breaking changes. Although you're right the message chain is also present in the verification Buffer.

For some reason this is also the case for the [NULS account](https://github.com/aleph-im/aleph-sdk-ts/blob/main/src/accounts/nuls.ts#L163) class in the Typescript SDK. So there's that... :man_shrugging: